modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -1999,6 +1999,16 @@ cargo::optional<Error> Builder::create<OpFunction>(const OpFunction *op) {
       module.addName(op->IdResult(), name);
     }
 
+    // DPC++ rejects variadic functions in SYCL code, with the exception of
+    // __builtin_printf which is accepts and generates invalid SPIR-V for. Patch
+    // this up.
+    if (name == "printf" && function_type->getNumParams() == 1 &&
```

Comment:
This says printf, but the comment says __builtin_printf

Reply:
The Clang built-in function `__builtin_printf` is translated into a call to the standard library function `printf`, and it's the latter we can detect. Suggestions welcome on how to make it clearer.

Reply:
Maybe just say that the __builtin_printf maps to printf.

Reply:
Thanks, added that with a bit more detail both here and in the commit message.